data-prepper-plugins/csv-processor/src/test/java/com/amazon/dataprepper/plugins/processor/csv/CSVProcessorTest.java

```diff
@@ -6,34 +6,316 @@
 package com.amazon.dataprepper.plugins.processor.csv;
 
 import com.amazon.dataprepper.metrics.PluginMetrics;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import com.amazon.dataprepper.model.record.Record;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 
-import static org.junit.jupiter.api.Assertions.assertThrows;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
 
 @ExtendWith(MockitoExtension.class)
 public class CSVProcessorTest {
-
     @Mock
     private CSVProcessorConfig processorConfig;
 
     @Mock
     private PluginMetrics pluginMetrics;
 
+    private CSVProcessor csvProcessor;
+
     @BeforeEach
     void setup() {
+        CSVProcessorConfig defaultConfig = new CSVProcessorConfig();
+        lenient().when(processorConfig.getSource()).thenReturn(defaultConfig.getSource());
+        lenient().when(processorConfig.getDelimiter()).thenReturn(defaultConfig.getDelimiter());
+        lenient().when(processorConfig.isDeleteHeader()).thenReturn(defaultConfig.isDeleteHeader());
+        lenient().when(processorConfig.getQuoteCharacter()).thenReturn(defaultConfig.getQuoteCharacter());
+        lenient().when(processorConfig.getColumnNamesSourceKey()).thenReturn(defaultConfig.getColumnNamesSourceKey());
+        lenient().when(processorConfig.getColumnNames()).thenReturn(defaultConfig.getColumnNames());
 
+        csvProcessor = createObjectUnderTest();
     }
 
     private CSVProcessor createObjectUnderTest() {
         return new CSVProcessor(pluginMetrics, processorConfig);
     }
 
     @Test
-    public void test_when_CSVProcessor_created_then_throws_NotImplemented_exception() {
-        assertThrows(UnsupportedOperationException.class, () -> createObjectUnderTest());
+    public void test_when_messageIsEmpty_then_notParsed() {
+        final Record<Event> eventUnderTest = createMessageEvent("");
+        final List<Record<Event>> editedEvents = (List<Record<Event>>) csvProcessor.doExecute(Collections.singletonList(eventUnderTest));
+        final Event parsedEvent = getSingleEvent(editedEvents);
+        assertThat(parsedEvent.containsKey("message"), equalTo(true));
+        assertThat(parsedEvent.containsKey("column1"), equalTo(false));
+    }
+
+    @Test
+    public void test_when_delimiterIsTab_then_parsedCorrectly() {
+        when(processorConfig.getDelimiter()).thenReturn("\t");
+        csvProcessor = createObjectUnderTest();
```

Comment:
You don't have to create this object in every test since it's been created in `setup()` method which is annotated with `@BeforeEach`.